src/meshes/t8code_mesh.jl

```diff
@@ -1117,3 +1121,456 @@ end
 @deprecate T8codeMesh{2}(meshfile::String; kwargs...) T8codeMesh(meshfile::String, 2; kwargs...)
 @deprecate T8codeMesh{3}(meshfile::String; kwargs...) T8codeMesh(meshfile::String, 3; kwargs...)
 #! format: on
+
+# Write the forest as vtu and also write the element's volumes in the file.
+#
+# t8code supports writing element based data to vtu as long as its stored
+# as doubles. Each of the data fields to write has to be provided in its own
+# array of length num_local_elements.
+# We support two types: T8_VTK_SCALAR - One double per element.
+#                  and  T8_VTK_VECTOR - Three doubles per element.
+function output_data_to_vtu(mesh::T8codeMesh, equations, solver,
+                            u_tmp, out)
+    vars = varnames(cons2cons, equations)
+
+    vtk_data = Vector{t8_vtk_data_field_t}(undef, nvariables(equations))
+
+    data = Array{Float64}(undef, ncells(mesh), nvariables(equations))
+    for v in eachvariable(equations)
+        for element in 1:ncells(mesh)
+            data[element, v] = u_tmp[element].u[v]
+        end
+    end
+
+    GC.@preserve data begin
+        for v in eachvariable(equations)
+            data_ptr = pointer(@views(data[:, v]))
+            vtk_data[v] = t8_vtk_data_field_t(T8_VTK_SCALAR,
+                                              NTuple{8192, Cchar}(rpad("$(vars[v])\0",
+                                                                       8192, ' ')),
+                                              data_ptr)
+        end
+
+        # The number of user defined data fields to write.
+        num_data = length(vtk_data)
+
+        # Write user defined data to vtu file.
+        write_treeid = 1
+        write_mpirank = 1
+        write_level = 1
+        write_element_id = 1
+        write_ghosts = 0
+        t8_forest_write_vtk_ext(mesh.forest, out, write_treeid, write_mpirank,
+                                write_level, write_element_id, write_ghosts,
+                                0, 0, num_data, pointer(vtk_data))
+    end
+end
+
+# Simple meshes
+# Temporary routines to create simple `cmesh`s by hand
+
+# Directly ported from: `src/t8_cmesh/t8_cmesh_examples.c: t8_cmesh_new_periodic_hybrid`.
+function cmesh_new_periodic_hybrid(comm)::t8_cmesh_t
+    n_dims = 2
+    vertices = [ # Just all vertices of all trees. partly duplicated
+        -1.0, -1.0, 0, # tree 0, triangle
+        0, -1.0, 0,
+        0, 0, 0,
+        -1.0, -1.0, 0, # tree 1, triangle
```

Comment:
The following 400 lines are just by-hand building routines for simple `cmesh`s. This is only temporary and will be deleted in the future.